src/models/data.cpp

```diff
@@ -389,3 +409,23 @@ const Data::ThreadEvents* Data::EventResults::findThread(qint32 pid, qint32 tid)
 {
     return const_cast<Data::EventResults*>(this)->findThread(pid, tid);
 }
+
+BottomUpResults BottomUpResults::diffBottomUpResults(const BottomUpResults& a, const BottomUpResults& b)
+{
+    if (a.costs.numTypes() != b.costs.numTypes()) {
+        return {};
```

Comment:
good catch! a warning should be emitted in this case at the very least such that we can see why this happens

furthermore, not only the number of types needs to be checked, but the actual types too - we don't want to diff one file that got recorded with `-e branches` with another one that got recorded with `-e cycles` e.g.

Reply:
this isn't done yet

Reply:
Parsing now fails if the such a situation is encountered.

Reply:
meaning this should become an assertion then?

Reply:
the perfparser calls diffBottomUpResults and if it receives an empty object parsingFailed will be emited